dandi/dandiapi.py

```diff
@@ -705,6 +705,17 @@ def get_raw_metadata(self) -> Dict[str, Any]:
         else:
             return cast(Dict[str, Any], self.client.get(self.api_path))
 
+    def get_blob_url(self) -> str:
+        """
+        Returns a non-redirecting URL for downloading the asset, found by
+        inspecting the metadata.  Raises `NotFoundError` if the metadata does
+        not contain an appropriate URL.
+        """
+        for url in self.get_raw_metadata().get("contentUrl", []):
+            if not url.startswith(self.client.api_url):
+                return url
```

Comment:
not sure if such approach would survive for long enough for the target use case (ROS3 access on S3)
- if we start populating with URLs which point elsewhere else, the first non-API one would not necessarily the one to use
- docstring suggests "non-redirecting URL" is returned, but there is no guarantee that those other non-API urls are non-redirecting. But then if we are to test for that -- we might just follow the redirects as I had suggested in #674

Reply:
So you want the function to just see where the download URL redirects to and return that?

Reply:
thought about it... may be let'just refine this one to 
```
def get_content_url(self, regex: str = '.*', follow_redirects=False) -> str:
```

- the concept of `blob` is specific to current implementation.  Later "content" might be a folder/prefix for a .zarr directory etc
- metadata has `contentUrl` so this would parallel nicely
- blob might reside not on AWS S3 so even with redirect, users expecting a URL to go to amazon would get some other, not the one they want

then for ROS3 (attn @bendichter ) could explicitly state their desire one way or another, e.g.
```
.get_content_url(follow_redirects=True)
```
and get a URL which no longer redirects, hopefully based on the fact that first url is the `.download_url` or even better more specific
```
.get_content_url('s3.amazonaws.com/.*blobs/', follow_redirects=True)
```
thus leaving no chance to a "chance": it would be guaranteed to be non-redirecting, and pointing to amazon S3 blobs collection.  
Since it will be a common pattern, we then add `.s3_blob_url` property (I am Ok with properties doing some work, but if you have strong objection @jwodder -- please give rationale) shortcut for the latter invocation.

Please also add a test for this functionality based on `000027` `sub-RAT123/sub-RAT123.nwb` - we have already tests relying on that asset to exist, so I think it is ok.

Reply:
@yarikoptic When `follow_redirects` is true, should matching against the regex occur before or after resolving redirects?  Either way, if multiple URLs match the given regex, which one should be returned?

Reply:
matching -- before. return only the first one like you do now in this PR.

Reply:
Should the test just fetch the default content URL, or is the S3 URL constant enough to rely on in tests?

Reply:
I would first check the url and then add download -- that file is tiny and having such integration testing would be useful for `dandi-api` integration testing against `dandi-cli`